spec/services/process_refund_service_spec.rb

```diff
@@ -4,22 +4,31 @@
 
 RSpec.describe ProcessRefundService do
   describe ".run" do
-    let(:finance_details) { double(:finance_details) }
+    let(:finance_details) { double(:finance_details, balance: -500) }
     let(:payment) { double(:payment, order_key: "123", registration_reference: "registration_reference", amount: 1_500) }
     let(:user) { double(:user, email: "user@example.com") }
     let(:payments) { double(:payments) }
     let(:refund) { double(:refund) }
+    let(:worldpay) { true }
 
     before do
       allow(payment).to receive(:worldpay?).and_return(worldpay)
     end
 
+    context "when the amount to refund is 0" do
```

Comment:
I think I'd benefit from tweaking the context title. We _know_ the amount to refund is 0 only because we know logic inside the service will calculate it like that because we have a positive balance.

So to me what we are actually testing is that if the balance is not negative it returns false and does not create a payment.

```suggestion
    context "when the registration balance is not negative" do
```